foreman/data_refinery_foreman/surveyor/array_express.py

```diff
@@ -42,54 +42,30 @@ class ArrayExpressSurveyor(ExternalSourceSurveyor):
     def source_type(self):
         return Downloaders.ARRAY_EXPRESS.value
 
-    def create_experiment_from_api(self, experiment_accession_code: str) -> (Experiment, Dict):
-        """Given an experiment accession code, create an Experiment object.
-
-        Also returns a dictionary of additional information about the
-        platform discovered for the experiment.
-
-        Will raise an UnsupportedPlatformException if this experiment was
-        conducted using a platform which we don't support.
-
-        See an example at: https://www.ebi.ac.uk/arrayexpress/json/v3/experiments/E-MTAB-3050/sample
-        """
-        request_url = EXPERIMENTS_URL + experiment_accession_code
-        experiment_request = utils.requests_retry_session().get(request_url, timeout=60)
-
-        try:
-            parsed_json = experiment_request.json()["experiments"]["experiment"][0]
-        except KeyError:
-            logger.error("Remote experiment has no Experiment data!",
-                         experiment_accession_code=experiment_accession_code,
-                         survey_job=self.survey_job.id)
-            raise
-
+    @staticmethod
+    def _apply_metadata_to_experiment(experiment_object: Experiment, parsed_json: Dict):
         experiment = {}
         experiment["name"] = parsed_json["name"]
-        experiment["experiment_accession_code"] = experiment_accession_code
+        experiment["experiment_accession_code"] = experiment_object.accession_code
 
-        # This experiment has no platform at all, and is therefore useless.
-        if 'arraydesign' not in parsed_json or len(parsed_json["arraydesign"]) == 0:
-            logger.warn("Remote experiment has no arraydesign listed.",
-                        experiment_accession_code=experiment_accession_code,
-                        survey_job=self.survey_job.id)
-            raise UnsupportedPlatformException
         # If there is more than one arraydesign listed in the experiment
         # then there is no other way to determine which array was used
         # for which sample other than looking at the header of the CEL
         # file. That obviously cannot happen until the CEL file has been
         # downloaded so we can just mark it as UNKNOWN and let the
         # downloader inspect the downloaded file to determine the
         # array then.
-        elif len(parsed_json["arraydesign"]) != 1 or "accession" not in parsed_json["arraydesign"][0]:
+        if len(parsed_json["arraydesign"]) != 1\
```

Comment:
You seem to be removing the check that parsed_json actually contains a key called `arraydesign`. I don't know how consistent their json is, so I'd prefer to keep such safety checks in place.

Reply:
Oh wait I see, you moved it to another place :)